### PR TITLE
janitorial: avoid one boolean argument

### PR DESCRIPTION
**Pull request by dankenigsberg**
This PR replaces a function with a boolean argument with two properly-named functions.

Function calls with a boolean argument, such as `Dump(duration, true)` are hard to read and understand. I believe that `DumpAllNamespaces(duration)` is clearer

```release-note
NONE
```

### CODE REVIEW

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -87,12 +87,20 @@ func (r *KubernetesReporter) SpecDidComplete(specSummary *types.SpecSummary) {
...
+func (r *KubernetesReporter) DumpTestNamespaces(duration time.Duration) {
+	r.dumpNamespaces(duration, tests.TestNamespaces)
+}
+
+func (r *KubernetesReporter) DumpAllNamespaces(duration time.Duration) {
+	r.dumpNamespaces(duration, []string{v1.NamespaceAll})
+}
+
+// Dump dumps the current state of the cluster. The relevant logs from the
+// specified duration are collected, limited to the specified vmiNamespaces.
```
Comment (EdDev) [inline tests/reporter/kubernetes.go:102]: nit: Documentation is done on the exposed functions usually.
And the convention is for the name of the method/function to be the first in the text.

**Review by EdDev — COMMENTED** [review]
Just one nit about documenting the function.

File: tests/reporter/kubernetes.go (hunk 1 above)
Comment (dankenigsberg) [reply tests/reporter/kubernetes.go:102]: I'd very happily drop the doc of the private function. But do I *have* to document the new functions I define in this PR? To me, `DumpAllNamespaces` is self explanatory; docs have a tendency to rot and confuse (as the first commit in this PR shows).

Comment (EdDev) [reply tests/reporter/kubernetes.go:102]: I personally do not mind, and this is not a formal library API which someone is suppose to consume (as a library).
So I am fine with removing it.

**Review by EdDev — APPROVED** [review]
Thanks!

**Review by davidvossel — COMMENTED** [review]
/approve

**Pull request merged**